src/property.ts

```diff
@@ -1020,6 +1022,12 @@ function Property$shouldSetValue(property: Property, obj: Entity, old: any, val:
 		throw new Error("Cannot set " + property.name + "=" + (val === undefined ? "<undefined>" : val) + " for instance " + obj.meta.type.fullName + "|" + obj.meta.id + ": a value of type " + (isEntityType(property.propertyType) ? property.propertyType.meta.fullName : parseFunctionName(property.propertyType)) + " was expected.");
 	}
 
+	for (let rule of property.rules) {
+		if (rule instanceof AllowedValuesRule && rule.preventInvalidValues && !rule.values(obj).includes(val) && !!val) {
```

Comment:
`!!val` would incorrectly skip validation of setting the property value to `false`. I think this is intended to be a `null/undefined` check?

Reply:
Yes, change made to explicitly check null/undefined